### ISSUE TITLE
Installation problem of Pyez on Windows

### ISSUE THREAD

**Issue by mahmutaydin1** [Juniper/py-junos-eznc#1077]
Hello Dear All,

I am trying to install Pyez module. I am using pycharm and trying to install from pycharm this  junos-eznc. However, I am receving some error. 

could you please help me for this problem?

C:\Users\asus\PycharmProjects\Junos\venv\Scripts\python.exe C:/Users/asus/PycharmProjects/Junos/pyez.py
Traceback (most recent call last):
  File "C:/Users/asus/PycharmProjects/Junos/pyez.py", line 1, in <module>
    import jnpr.junos
  File "C:\Users\asus\PycharmProjects\Junos\venv\lib\site-packages\jnpr\junos\__init__.py", line 3, in <module>
    from jnpr.junos.factory.to_json import PyEzJSONEncoder
  File "C:\Users\asus\PycharmProjects\Junos\venv\lib\site-packages\jnpr\junos\factory\__init__.py", line 4, in <module>
    from jnpr.junos.factory.factory_loader import FactoryLoader
  File "C:\Users\asus\PycharmProjects\Junos\venv\lib\site-packages\jnpr\junos\factory\factory_loader.py", line 14, in <module>
    from jnpr.junos.factory.factory_cls import *
  File "C:\Users\asus\PycharmProjects\Junos\venv\lib\site-packages\jnpr\junos\factory\factory_cls.py", line 7, in <module>
    from jnpr.junos.factory.cmdtable import CMDTable
  File "C:\Users\asus\PycharmProjects\Junos\venv\lib\site-packages\jnpr\junos\factory\cmdtable.py", line 19, in <module>
    from ntc_templates import parse as ntc_parse
  File "C:\Users\asus\PycharmProjects\Junos\venv\lib\site-packages\ntc_templates\parse.py", line 3, in <module>
    from textfsm import clitable
  File "C:\Users\asus\PycharmProjects\Junos\venv\lib\site-packages\textfsm\clitable.py", line 41, in <module>
    from textfsm import texttable
  File "C:\Users\asus\PycharmProjects\Junos\venv\lib\site-packages\textfsm\texttable.py", line 42, in <module>
    from textfsm import terminal
  File "C:\Users\asus\PycharmProjects\Junos\venv\lib\site-packages\textfsm\terminal.py", line 25, in <module>
    import fcntl
ModuleNotFoundError: No module named 'fcntl'

**Comment by ntwrkguru** [comment]
Please see issue #1015 for resolution involving changing the version of textfsm. Also, this is fixed in any commits after #1019, so consider updating the library.

**Comment by mahmutaydin1** [comment]
Hi
ı installed textfsm. After that, when try to install junos-eznc, I receive below error.

Collecting junos-eznc
  Using cached https://files.pythonhosted.org/packages/2d/56/3a83765eb458b56725f254e80195d39151d32cfe45ea57c4deaa327fda83/junos_eznc-2.5.3-py2.py3-none-any.whl
Collecting ncclient>=0.6.3 (from junos-eznc)
Collecting jinja2>=2.7.1 (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/30/9e/f663a2aa66a09d838042ae1a2c5659828bb9b41ea3a6efa20a20fd92b121/Jinja2-2.11.2-py2.py3-none-any.whl
Requirement already satisfied: six in c:\users\asus\pycharmprojects\untitled\venv\lib\site-packages (from junos-eznc) (1.15.0)
Collecting ntc-templates (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/32/1a/4d0927a0f8093d3df65d2768f9ab813867bcba5596a6c10100e346f60c75/ntc_templates-1.5.0-py3-none-any.whl
Collecting scp>=0.7.0 (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/4d/7a/3d76dc5ad8deea79642f50a572e1c057cb27e8b427f83781a2c05ce4e5b6/scp-0.13.2-py2.py3-none-any.whl
Collecting PyYAML>=5.1 (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/64/c2/b80047c7ac2478f9501676c988a5411ed5572f35d1beff9cae07d321512c/PyYAML-5.3.1.tar.gz
Collecting paramiko>=1.15.2 (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/95/19/124e9287b43e6ff3ebb9cdea3e5e8e88475a873c05ccdf8b7e20d2c4201e/paramiko-2.7.2-py2.py3-none-any.whl
Collecting pyserial (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/0d/e4/2a744dd9e3be04a0c0907414e2a01a7c88bb3915cbe3c8cc06e209f59c30/pyserial-3.4-py2.py3-none-any.whl
Collecting pyparsing (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/8a/bb/488841f56197b13700afd5658fc279a2025a39e22449b7cf29864669b15d/pyparsing-2.4.7-py2.py3-none-any.whl
Collecting lxml>=3.2.4 (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/2c/4d/3ec1ea8512a7fbf57f02dee3035e2cce2d63d0e9c0ab8e4e376e01452597/lxml-4.5.2.tar.gz
Collecting transitions (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/a6/c9/dcdd9fb8d325df26644793fc39c2eecf15692d277e39bcf52e6dd05fffb1/transitions-0.8.2-py2.py3-none-any.whl
Collecting netaddr (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/ff/cd/9cdfea8fc45c56680b798db6a55fa60a22e2d3d3ccf54fc729d083b50ce4/netaddr-0.8.0-py2.py3-none-any.whl
Collecting yamlordereddictloader (from junos-eznc)
  Using cached https://files.pythonhosted.org/packages/56/e1/1ca77da64cc355f0de483095e841d96f2366f93b095b83869440a296c21d/yamlordereddictloader-0.4.0.tar.gz
Requirement already satisfied: setuptools>0.6 in c:\users\asus\pycharmprojects\untitled\venv\lib\site-packages\setuptools-39.1.0-py3.8.egg (from ncclient>=0.6.3->junos-eznc) (39.1.0)
Collecting MarkupSafe>=0.23 (from jinja2>=2.7.1->junos-eznc)
  Using cached https://files.pythonhosted.org/packages/b9/2e/64db92e53b86efccfaea71321f597fa2e1b2bd3853d8ce658568f7a13094/MarkupSafe-1.1.1.tar.gz
Requirement already satisfied: textfsm>=1.1.0 in c:\users\asus\pycharmprojects\untitled\venv\lib\site-packages (from ntc-templates->junos-eznc) (1.1.0)
Collecting bcrypt>=3.1.3 (from paramiko>=1.15.2->junos-eznc)
  Using cached https://files.pythonhosted.org/packages/d8/ba/21c475ead997ee21502d30f76fd93ad8d5858d19a3fad7cd153de698c4dd/bcrypt-3.2.0.tar.gz

  Could not find a version that satisfies the requirement cffi>=1.1 (from versions: )
No matching distribution found for cffi>=1.1
You are using pip version 10.0.1, however version 20.2.3 is available.
You should consider upgrading via the 'python -m pip install --upgrade pip' command.

**Comment by ntwrkguru** [comment]
What version of Python are you using? Seems pretty old to have pip 10.0.1.

**Comment by vnitinv** [comment]
I am not able to replicate this @mahmutaydin1. As suggested by @ntwrkguru, try upgrading pip and let us now if you still face issue.
Also share pip list before running install for PyEZ.

**Comment by mahmutaydin1** [comment]
upgrade to pip is not working guys.

**Comment by ntwrkguru** [comment]
@mahmutaydin1 Please post the output of `pip list` as well as `python -V`.

**Comment by rahkumar651991** [comment]
@mahmutaydin1 - closing this issue for now as it has been some time with no further discussion. 
If issue still arrives you may re-open it or ope a new issue.
